### PR TITLE
Dashboard: Add resiliency around malformed stories data

### LINKED ISSUE #6094: Dashboard: Add resiliency if story_data is missing
## Feature Description

It appears that some users are having troubles loading the dashboard if their story data is somehow corrupt or missing. The dashboard crashes entirely if for example the `story_data` field is empty.

To mitigate this, let's add some simple resiliency around that to prevent crashes.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

In some environments, `story_data` can be empty or missing, causing the dashboard to crash.

## Summary

This PR avoids crashes by skipping stories without the expected data format.

## To-do

* [x] Display error instead of just silently skipping errors

## Testing Instructions

Add the following on line 125 of `Stories_Base_Controller.php`:

```php
$data['story_data'] = [];
```

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6094

### CODE REVIEW

**Comment by swissspidy** [comment]
Perhaps could just detect anomaly & set error state like this:

```js
if (Object.values(reshapedStories).length !== action.payload.stories.length) {
  return {
    ...state,
    error: { ...action.payload, id: Date.now() },
  };
}
````

File: assets/src/dashboard/app/reducer/stories.js [hunk 1]
```diff
@@ -151,7 +151,9 @@ function storyReducer(state, action) {
     case ACTION_TYPES.FETCH_STORIES_SUCCESS: {
       const fetchedStoriesById = [];
       const reshapedStories = action.payload.stories.reduce((acc, current) => {
-        if (!current) {
+        // Hardening in case data returned by the server is malformed.
+        // For example, story_data could be missing.
+        if (!current || !current?.story_data?.length) {
```
Comment (BrittanyIRL) [inline assets/src/dashboard/app/reducer/stories.js]: This actually feels like something that should be caught before getting to the reducer. What about this in the `useStoryApi` - around line 104. 

```
 const cleanStories = response.body.filter(
          (story) => story?.story_data?.length > 0
        );

        if (cleanStories.length <= 0) {
          dispatch({
            type: STORY_ACTION_TYPES.FETCH_STORIES_FAILURE,
            payload: {
              message: {
                body: _n('Story Data Missing', 'web-stories'),
                title: ERRORS.LOAD_STORIES.TITLE,
              },
            },
          });
        } else {
          dispatch({
            type: STORY_ACTION_TYPES.FETCH_STORIES_SUCCESS,
            payload: {
              editStoryURL,
              stories: cleanStories,
              totalPages,
              totalStoriesByStatus: {
                ...totalStoriesByStatus,
                [STORY_STATUS.PUBLISHED_AND_FUTURE]:
                  totalStoriesByStatus[STORY_STATUS.PUBLISH] +
                  totalStoriesByStatus[STORY_STATUS.FUTURE],
              },
              page,
            },
          });
        }
      } catch (err) {
...
```

Comment (swissspidy) [reply assets/src/dashboard/app/reducer/stories.js]: Ah yes that looks much cleaner 👍

**Comment by swissspidy** [comment]
The only downside here is that if 10 out of 20 stories are malformed, the user will only see 10 without any explanation. But that's definitely better than the current situation where the dashboard crashes.

File: assets/src/dashboard/app/api/useStoryApi.js [hunk 2]
```diff
@@ -102,11 +102,33 @@ const useStoryApi = (dataAdapter, { editStoryURL, storyApi, encodeMarkup }) => {
           response.headers &&
           JSON.parse(response.headers['X-WP-TotalByStatus']);
 
+        // Hardening in case data returned by the server is malformed.
+        // For example, story_data could be missing/empty.
+        const cleanStories = response.body.filter((story) =>
+          Array.isArray(story?.story_data?.pages)
```
Comment (BrittanyIRL) [inline assets/src/dashboard/app/api/useStoryApi.js:108]: 👍

**Review by BrittanyIRL — APPROVED** [review]
Thank you for fixing this!

**Pull request merged**